src/hotspot/share/gc/g1/g1RootClosures.cpp

```diff
@@ -50,10 +50,10 @@ class G1EvacuationClosures : public G1EvacuationRootClosures {
 // Closures used during concurrent start.
 // The treatment of "weak" roots is selectable through the template parameter,
 // this is usually used to control unloading of classes and interned strings.
-template <G1Mark MarkWeak>
+template <bool mark_weak>
```

Comment:
I think it could make sense to name this parameter `should_mark_weak` to be more consistent. But if you don't agree just leave it.

Reply:
Fixed.